src/crimson/os/seastore/lba_manager/btree/btree_lba_manager.h

```diff
@@ -61,9 +63,14 @@ using LBABtree = FixedKVBtree<
  * get_mappings, alloc_extent_*, etc populate a Transaction
  * which then gets submitted
  */
+template <bool leaf_has_children>
```

Comment:
Continue the thread https://github.com/ceph/ceph/pull/47749#discussion_r1119517739

> This is because leaf nodes in LBA unit tests don't have child nodes and a lot of assertion checks in the code don't hold.
> 
> The reason why I used templates for this is that, if not templates, probably I need to add a field in LBALeafNode to do the distinguishing or use a macro for this, the former would cause cpu overhead when running in normal cases and the latter may lead to a coding style pretty like the "template" way(macro everywhere VS template everywhere).

How about complementing the LBA unit test with children (test extents) instead, rather than specializing the normal paths?

My major concern is about the pervasive specializations which are just to pass some tests. This template is infectious and will only be more complicated in the future. And there is still a major feature gap in the FixedKVBtree to support cloning.

Reply:
Ok, I'll try this way.